polaris/dataset/_competition_dataset.py

```diff
@@ -0,0 +1,56 @@
+from typing import Optional
+
+from pydantic import model_validator
+from polaris.dataset import Dataset
+from polaris.utils.constants import DEFAULT_CACHE_DIR
+from polaris.utils.errors import InvalidDatasetError, PolarisChecksumError
+from datamol.utils import fs
+from polaris.dataset._column import ColumnAnnotation
+
+_CACHE_SUBDIR = "datasets"
+
+
+class CompetitionDataset(Dataset):
+    masked_md5sum: Optional[str] = None
+
+    ## CHECK IF THIS
+    @model_validator(mode="after")
+    @classmethod
+    def _validate_model(cls, m: "CompetitionDataset"):
+        """If a checksum is provided, verify it matches what the checksum should be for the masked dataset.
+        If no checksum is provided, make sure it is set.
+        If no cache_dir is provided, set it to the default cache dir and make sure it exists
+        """
+
+        # Verify that all annotations are for columns that exist
+        if any(k not in m.table.columns for k in m.annotations):
+            raise InvalidDatasetError("There are annotations for columns that do not exist")
+
+        # Verify that all adapters are for columns that exist
+        if any(k not in m.table.columns for k in m.default_adapters.keys()):
+            raise InvalidDatasetError("There are default adapters for columns that do not exist")
+
+        # Set a default for missing annotations and convert strings to Modality
+        for c in m.table.columns:
+            if c not in m.annotations:
+                m.annotations[c] = ColumnAnnotation()
+            m.annotations[c].dtype = m.table[c].dtype
+
+        # Verify the checksum
+        actual = m.masked_md5sum
```

Comment:
This is the difference with regular datasets -- competition datasets have certain masked values that need to be accounted for.